server/src/test/java/org/opensearch/index/store/remote/file/OnDemandBlockSnapshotIndexInputTests.java

```diff
@@ -106,7 +106,7 @@ public void testChunkedRepository() throws IOException {
             indexInput.seek(repositoryChunkSize);
         }
         // Verify the second chunk is requested (i.e. ".part1")
-        verify(transferManager).fetchBlob(argThat(request -> request.getBlobName().equals("File_Name.part1")));
+        verify(transferManager).fetchBlob(argThat(request -> request.blobParts().get(0).getBlobName().equals("File_Name.part1")));
     }
 
     private void runAllTestsFor(int blockSizeShift) throws Exception {
```

Comment:
All the test scenarios in here passed before you made this change (i.e. repository chunk size was a multiple of the block size). We should add some scenarios where the tests would have failed without your change.

Reply:
Sure. Added in latest commit